src/eko/io/struct.py

```diff
@@ -327,115 +326,109 @@ def deepcopy(self, path: os.PathLike):
-        new.access.path = pathlib.Path(path)
+        new.access.path = path
         new.access.readonly = False
         new.access.open = True
 
-        tmpdir = pathlib.Path(tempfile.mkdtemp(prefix=TEMP_PREFIX))
+        tmpdir = Path(tempfile.mkdtemp(prefix=TEMP_PREFIX))
         new.metadata.path = tmpdir
         # copy old dir to new dir
         tmpdir.rmdir()
         shutil.copytree(self.paths.root, new.paths.root)
         new.close()
 
     @staticmethod
-    def load(tarpath: os.PathLike, dest: os.PathLike):
-        """Load the content of archive in a target directory.
-
-        Parameters
-        ----------
-        tarpath: os.PathLike
-            the archive to extract
-        tmppath: os.PathLike
-            the destination directory
-
-        """
+    def extract(tarpath: Path, dest: Path):
+        """Extract the content of archive in a target directory."""
         try:
             with tarfile.open(tarpath) as tar:
                 raw.safe_extractall(tar, dest)
         except tarfile.ReadError:
             raise exceptions.OutputNotTar(f"Not a valid tar archive: '{tarpath}'")
 
     @classmethod
-    def open(cls, path: os.PathLike, mode="r"):
-        """Open EKO object in the specified mode."""
-        path = pathlib.Path(path)
-        access = AccessConfigs(path, readonly=False, open=True)
-        load = False
-        if mode == "r":
-            load = True
-            access.readonly = True
-        elif mode in "w":
-            pass
-        elif mode in "a":
-            load = True
-        else:
-            raise ValueError(f"Unknown file mode: {mode}")
-
-        tmpdir = pathlib.Path(tempfile.mkdtemp(prefix=TEMP_PREFIX))
-        if load:
-            cls.load(path, tmpdir)
-            metadata = Metadata.load(tmpdir)
-            opened = cls(
-                **inventories(tmpdir, access),
-                metadata=metadata,
-                access=access,
-            )
-            opened.operators.sync()
-        else:
-            opened = Builder(path=tmpdir, access=access)
+    def load(cls, path: Path):
+        """Load the EKO from disk information.
 
-        return opened
+        Note
+        ----
+        No archive path is assigned to the :cls:`EKO` object, setting its
+        :attr:`EKO.access.path` to `None`.
+        If you want to properly load from an archive, use the :meth:`read`
+        constructor.
+
+        """
+        access = AccessConfigs(None, readonly=True, open=True)
+
+        metadata = Metadata.load(path)
+        loaded = cls(
+            **inventories(path, access),
+            metadata=metadata,
+            access=access,
+        )
+        loaded.operators.sync()
+
+        return loaded
 
     @classmethod
-    def read(cls, path: os.PathLike):
-        """Read the content of an EKO.
+    def read(
+        cls,
+        path: Path,
+        extract: bool = True,
+        dest: Optional[Path] = None,
+        readonly: bool = True,
+    ):
```

Comment:
This looks to complicated ... can we make `from_tar` and `from_dir` instead (if you wish keep this low-level) and declare `extract` public?

Reply:
First, I'd like to avoid the `from_` constructors altogether. There is a single EKO format, that's why `read` is useful.

Then, the EKO format is simple, there is nothing so complicate. If you want to extract you do not need `EKO.extract("a.tar", "b/")`, but you can go directly for `tar = tarfile.open("a.tar"); tar.extractall("b/")`. If we are just using the standard library for some operation, there is no need to hide it.
In practice, I expect `EKO.read("a.tar", dest="b/"))` to be used exactly in our use case: you want to open a folder in a controller for many agents to act on it. In principle, it could be even inside the context manager, if you're spawning the process from there, or even just waiting for their completion (and it would be recommended, at least someone will eventually close it).

So, no need for `EKO.extract()` in any case.
The `from_tar` is just `read`, because most of the time you will just read a tar, and nothing else. So it's only a renaming (but a name I don't like, gives the impression of many possible formats, but runtime and disk representation are coupled, so it's not really possible).
The `from_dir` is just an alias for `read(..., extract=False)`, so it's the exact same game of `edit`. However, if we play this game for both of them, we'll have to provide the `from_dir_edit`, and this I won't do. And I'd advocate for `edit`, rather than `from_dir`, since `edit` we are already using plenty of times.

Reply:
Sorry, I forgot one point.

> Re `extract`: I would keep the function (about the content I'm open) just in case we change the compression at some point (that being said the new function should maybe be called `from_archive`?)

There is no compression: `tar` is only there to make a single blob, but the EKO is the folder inside (that's why we kept the `.tar`).
The compression is at the level of the arrays, and it's internally managed by EKO. So this should never affect the end user directly.